pkg/apis/experimental/types.go

```diff
@@ -578,3 +578,64 @@ type IngressBackend struct {
 	// Specifies the port of the referenced service.
 	ServicePort util.IntOrString `json:"servicePort"`
 }
+
+type NodeResource string
+
+const (
+	// Percentage of node's CPUs that is currently used.
+	CpuConsumption NodeResource = "CpuConsumption"
+
+	// Percentage of node's CPUs that is currently requested for pods.
+	CpuRequest NodeResource = "CpuRequest"
+
+	// Percentage od node's memory that is currently used.
+	MemConsumption NodeResource = "MemConsumption"
+
+	// Percentage of node's CPUs that is currently requested for pods.
+	MemRequest NodeResource = "MemRequest"
+)
+
+// NodeUtilization describes what percentage of a particular resource is used on a node.
+type NodeUtilization struct {
+	Resource NodeResource `json:"resource"`
+
+	// The accepted values are from 0 to 1.
+	Value float64 `json:"value"`
+}
+
+// Configuration of the Cluster Autoscaler
+type ClusterAutoscalerSpec struct {
+	// Minimum number of nodes that the cluster should have.
+	MinNodes int `json:"minNodes"`
+
+	// Maximum number of nodes that the cluster should have.
+	MaxNodes int `json:"maxNodes"`
```

Comment:
I think that until we have heterogeneous cluster it's easier for the user to provide number of nodes instead of computing number of cores/amount of memory.